src/typings/http-client.d.ts

```diff
@@ -52,12 +53,8 @@ declare module "http-client" {
         withBaseUrl(url: string): HttpClientConfiguration;
     }
 
-    export class HttpApiClient {
-
-    }
-
     interface IHttpClientMiddleware {
         (config: IHttpClientRequestOptions, next: (options: IHttpClientRequestOptions) => Promise<any>):
-            void | IHttpClientRequestOptions | HttpClientResponse;
+            void | IHttpClientRequestOptions | HttpClientResponse<any>;
```

Comment:
The return type is wrong. This is not your error, but we should open up an issue to fix this. I believe it should just allow for Promise<HttpClientResponse<any>>. This should be changed for the return type of the *next*-parameter as well.

You can open up an issue for this and fix it within the same PullRequest.